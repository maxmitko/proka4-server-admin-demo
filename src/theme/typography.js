
const defaultFontFamily = "'Roboto', sans-serif"

export default {
    h1: {
        fontWeight: 300,
        fontSize: 6,
        letterSpacing: -1.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h2: {
        fontWeight: 300,
        fontSize: 3.75,
        letterSpacing: -0.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h3: {
        fontWeight: 400,
        fontSize: 3,
        letterSpacing: 0,
        lineHeight: 1.04,
        fontFamily: defaultFontFamily,
        tag: "h3",
    },
    h4: {
        fontWeight: 400,
        fontSize: 2.125,
        letterSpacing: 0.25,
        lineHeight: 1.17,
        fontFamily: defaultFontFamily,
        tag: "h4",
    },
    h5: {
        fontWeight: 400,
        fontSize: 1.5,
        letterSpacing: 0,
        lineHeight: 1.33,
        fontFamily: defaultFontFamily,
        tag: "h5",
    },
    h6: {
        fontWeight: 300,
        fontSize: 1.15,
        letterSpacing: 0,
        lineHeight: 1.4,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.15,
        lineHeight: 1.75,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle2: {
        fontWeight: 500,
        fontSize: 0.875,
        letterSpacing: 0.1,
        lineHeight: 1.57,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    body1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.5,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    body2: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 0.25,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    button: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 1.25,
        lineHeight: 1.5,
        fontFamily: "'Roboto Condensed', sans-serif",
        tag: "span",
    },
    caption: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 0.4,
        lineHeight: 1.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    overline: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 1.5,
        lineHeight: 2.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    inherit: {
        fontWeight: "inherit",
        fontSize: "inherit",
        letterSpacing: "inherit",
        lineHeight: "inherit",
        fontFamily: "inherit",
        tag: "span",
    },
}