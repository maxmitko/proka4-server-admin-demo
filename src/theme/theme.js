import typography from './typography'
import produce from "immer"

export const theme = {
    palette: {
        primary: "hsla(0, 0%, 0%, 1)",
        secondary: "hsla(187, 100%, 42%, 1)",
        background: "hsla(0, 0%, 93%, 1)",
        surface: "hsla(0, 0%, 100%, 1)",
        error: "hsla(0, 100%, 42%, 1)",
    },
    paletteOn: {
        primary: "hsla(0, 0%, 100%, 1)",
        secondary: "hsla(0, 0%, 100%, 1)",
        background: "hsla(0, 0%, 0%, 1)",
        surface: "hsla(210, 9%, 31%, 1)",
        error: "hsla(0, 0%, 100%, 1)",
    },
    extra: {
        default: null,
        gray: "hsla(0, 0%, 76%, 1)",
        primary: "hsla(291, 64%, 42%, 1)",
        danger: "hsla(4, 90%, 58%, 1)",
        success: "hsla(123, 36%, 52%, 1)",
        warning: "hsla(34, 98%, 51%, 1)",
        info: "hsla(187, 100%, 42%, 1)",
        rose: "hsla(340, 82%, 58%, 1)",
        white: "hsla(0, 0%, 100%, 1)",
    },
    extraOn: {
        default: null,
        gray: "hsla(0, 0%, 100%, 1)",
        primary: "hsla(0, 0%, 100%, 1)",
        danger: "hsla(0, 0%, 100%, 1)",
        success: "hsla(0, 0%, 100%, 1)",
        warning: "hsla(0, 0%, 100%, 1)",
        info: "hsla(0, 0%, 100%, 1)",
        rose: "hsla(0, 0%, 100%, 1)",
        white: "hsla(0, 0%, 0%, 1)",
    },
    extraLight: {
        default: null,
        gray: "hsla(0, 0%, 72%, 1)",
        primary: "hsla(291, 88%, 57%, 1)",
        danger: "hsla(22, 100%, 53%, 1)",
        success: "hsla(123, 45%, 57%, 1)",
        warning: "hsla(34, 98%, 60%, 1)",
        info: "hsla(187, 95%, 46%, 1)",
        rose: "hsla(340, 82%, 65%, 1)",
        white: "hsl(0, 100%, 100%)",
    },
    extraShadow: {
        default: null,
        gray: "hsla(0, 0%, 30%, 0.22)",
        primary: "hsla(291, 64%, 54%, 0.25)",
        danger: "hsla(0, 100%, 58%, 0.25)",
        success: "hsla(123, 45%, 57%, 0.25)",
        warning: "hsla(34, 98%, 60%, 0.25)",
        info: "hsla(187, 95%, 46%, 0.25)",
        rose: "hsla(340, 82%, 63%, 0.25)",
        white: "hsla(0, 0%, 30%, 0.25)",
    },
    typography: typography,
    zIndex: {
        mobileStepper: 1000,
        appBar: 1100,
        drawer: 1200,
        modal: 1300,
        snackbar: 1400,
        tooltip: 1500,
    }
}

const newTheme = produce(theme, draftTheme => {
    draftTheme.extra.default = theme.extra.primary
    draftTheme.extraOn.default = theme.extraOn.primary
    draftTheme.extraLight.default = theme.extraLight.primary
    draftTheme.extraShadow.default = theme.extraShadow.primary
})


export default newTheme