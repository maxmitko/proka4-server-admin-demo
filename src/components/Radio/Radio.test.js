import { mount } from 'enzyme';
import React from 'react';
import { ThemeProvider } from 'styled-components'
import theme from 'theme/theme'
import TestComponent from './index'
import renderer from 'react-test-renderer';
import produce from 'immer'

const props = {
    checked: true,
    disabled: false,
    color: 'default',
}

const withHOC = ui => (
    <ThemeProvider theme={theme}>
        {ui}
    </ThemeProvider >
)

describe('enzyme Radio', () => {
    let wrapper;
    let element;
    const dangerColor = theme.extra.danger.replace(/\s/g, '')

    beforeEach(() => {
        element = props => withHOC(<TestComponent {...props} />)
    })

    afterEach(() => {
        wrapper && wrapper.unmount();
    });

    it('default SNAPSHOT should render correctly', () => {
        const tree = renderer.create(element(props)).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('checked should render correctly', () => {
        wrapper = mount(element(props));
        expect(wrapper.find('div[checked=true] span').html()).toBeDefined();
    });
    
    it('unchecked should render correctly', () => {
        const newProps = produce(props, draftProps => {
            draftProps.checked = false
        })
        wrapper = mount(element(newProps));
        expect(() => { wrapper.find('div[checked=false] span').html() }).toThrow();
    });

    it('disabled should render correctly', () => {
        const newProps = produce(props, draftProps => {
            draftProps.disabled = true
        })
        wrapper = mount(element(newProps));
        expect(wrapper.find('div[checked=true]')).toHaveStyleRule("border-color", "#ccc");
        expect(wrapper.find('div[checked=true] span')).toHaveStyleRule("background-color", "#ccc");
    });
    
    it('change color should render correctly', () => {
        const newProps = produce(props, draftProps => {
            draftProps.color = 'danger'
        })
        wrapper = mount(element(newProps));

        expect(wrapper.find('div[checked=true]')).toHaveStyleRule("border-color", dangerColor);
        expect(wrapper.find('div[checked=true] span')).toHaveStyleRule("background-color", dangerColor);
    });

});
