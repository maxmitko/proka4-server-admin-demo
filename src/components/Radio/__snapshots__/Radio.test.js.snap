// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme Radio default SNAPSHOT should render correctly 1`] = `
.c1 {
  cursor: pointer;
  display: none;
}

.c2 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: hsla(291,64%,42%,1);
}

.c0 {
  border-radius: 50%;
  margin: 0 7px;
  border: 2px solid;
  box-sizing: border-box;
  border-color: hsla(291,64%,42%,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3px;
  min-width: 1.2rem;
  height: 1.2rem;
}

<div
  checked={true}
  className="c0"
  color="default"
  disabled={false}
>
  <input
    className="c1"
    disabled={false}
    type="checkbox"
  />
  <span
    className="c2"
    color="default"
    disabled={false}
  />
</div>
`;
